provider/aws/aws.go

```diff
@@ -218,13 +217,13 @@ type Route53Changes []*Route53Change
 
 type profiledZone struct {
 	profile string
-	zone    *route53.HostedZone
+	zone    *route53types.HostedZone
 }
 
-func (cs Route53Changes) Route53Changes() []*route53.Change {
-	ret := []*route53.Change{}
+func (cs Route53Changes) Route53Changes() []route53types.Change {
```

Comment:
Are you sure the `[]*` to `[]` change is fine?

Reply:
Very likely this will make https://github.com/kubernetes-sigs/external-dns/pull/4640/files#diff-010666439eeb90737961580952d89821bc8ef6cc69bced421af68dbcde0eaf29R57 never happen.
I would rather not change the ptr to a value, because there are too many unexpected checks that we could not see in your PR and miss these.
I suggest that you convert it to the same type (ptr/value) as in our original code and if you want to change ptr to value, then add another separated PR.

Reply:
> Are you sure the []* to [] change is fine?

The reason for this change is because in aws-sdk-go-v2, the `ChangeBatch` struct changed from needing a `[]*Change` to `[]Change`
* https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/route53@v1.42.3/types#ChangeBatch
* https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#ChangeBatch

and do think it's safe based on how it's being used, e.g. https://github.com/kubernetes-sigs/external-dns/commit/ea921690d22fa20329365dd3f645935104343954#diff-95fde44621b48d924a9ad43138223f86d3012e63c65400f7051b7cad08096052R641
---
> Very likely this will make https://github.com/kubernetes-sigs/external-dns/pull/4640/files#diff-010666439eeb90737961580952d89821bc8ef6cc69bced421af68dbcde0eaf29R57 never happen.

I may be misunderstanding this part of the second comment, but I think it's unrelated (`Change` vs `Config`). I agree in principal with the rest of your second comment, but hopefully the reasoning for this specific change makes sense given the context for this one. There have been a lot of ptr/value changes between aws-sdk-go and aws-sdk-go-v2. Happy to align other ptr/value changes if there's no need to change.

Reply:
ok